### PR TITLE
Feat: Improve interoperability on server-to-server calls+

### PR DESCRIPTION
**Pull request by brunomenezes**
### Summary
Code changes to address some feedback received after using Sunodo tech combined with the latest docker images of rollups-explorer and rollups-explorer-api. 

PR reference: 
* https://github.com/sunodo/sunodo/pull/244

Changes: 
* Fix the double encoding on the send-raw input form. The for was correctly enforcing the input to be a `hex` value (e.g. 0x4020) but also re-encoding it as hex before sending it as a transaction argument.
* Add optional environment variable to improve interoperability between servers. 
* A few DX improvements related to typescript autocompletion.

### CODE REVIEW

File: apps/web/additional.d.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+declare namespace NodeJS {
+    export interface ProcessEnv {
+        /**
+         * Network chain id
+         */
+        NEXT_PUBLIC_CHAIN_ID: string;
+        /**
+         * Public available rollups-xplorer-api graphql endpoint.
```
Comment (nevendyulgerov) [inline apps/web/additional.d.ts:8]: Typo here, should be `rollups-explorer-api`

File: packages/ui/src/RawInputForm.tsx [hunk 2]
```diff
@@ -53,7 +53,7 @@ export const RawInputForm: FC<RawInputFormProps> = (props) => {
             address: isAddress(values.application)
                 ? getAddress(values.application)
                 : zeroAddress,
-            rawInput: toHex(values.rawInput),
+            rawInput: values.rawInput as Hex,
```
Comment (nevendyulgerov) [inline packages/ui/src/RawInputForm.tsx:56]: Nice catch!

**Review by nevendyulgerov — APPROVED** [review]
LGTM!

File: packages/ui/src/EtherDepositForm.tsx [hunk 3]
```diff
@@ -71,7 +71,7 @@ export const EtherDepositForm: FC<EtherDepositFormProps> = (props) => {
                           chain?.nativeCurrency.decimals ?? 18,
                       )
                     : undefined,
-            execLayerData: toHex(values.execLayerData),
+            execLayerData: values.execLayerData as Hex,
```
Comment (nevendyulgerov) [inline packages/ui/src/EtherDepositForm.tsx:74]: I applied the same fix for the `EtherDepositForm`, because same issue was reproducible for it as well.

**Review by nevendyulgerov — APPROVED** [review]

**Comment by brunomenezes** [comment]
> LGTM! I took the liberty to commit a few more changes:
> 
> * Applied the same fix for the incorrect hex parsing to the `EtherDepositForm`
> * Covered this case with unit tests for both `RawInputForm` and `EtherDepositForm`.
> 
> I hope you don't mind @brunomenezes :)

No problem! you could have included the typos in that commit =). I did a `commit fixup`, and I will be merging since you approved. it is just a comment typo fix.

**Pull request merged**